src/v/archival/ntp_archiver_service.cc

```diff
@@ -566,15 +567,15 @@ bool ntp_archiver::can_update_archival_metadata() const {
 }
 
 ss::future<> ntp_archiver::stop() {
-    _uploads_active.broken();
-    _leader_cond.broken();
     if (_local_segment_merger) {
         if (!_local_segment_merger->interrupted()) {
             _local_segment_merger->interrupt();
         }
         co_await _local_segment_merger.get()->stop();
     }
     _as.request_abort();
+    _uploads_active.broken();
```

Comment:
Unrelated to this change, but for posterity, in coming back to this code after a bit, I was at first surprised the first thing we do _isn't_ request an abort from the abort source. Looks like the local segment merger has its own abort source, and there's no interdependence in stopping it.